### ISSUE TITLE
Problem with the Unzip type family

### ISSUE THREAD

**Issue by Lysxia** [tomjaguarpaw/product-profunctors#44]
`Unzip` assumes that in any type of the form `f a`, where `a` has kind `Type`, `a` is a type of field. But `a` could also be phantom or a parameter of a `data` family (or yet something else?).

Related to #42 because this another problem with `genericAdaptor`, but #42 might be fixed independently in a way that works for most common cases.

The present issue seems to require an overhaul of `Unzippable`, for which #42 should certainly be taken into account.

**Comment by tomjaguarpaw** [comment]
I wonder if `genericAdaptor` is not really a well-defined thing.  It seems somewhat challenging to write down a mathematical definition of its behaviour.

**Comment by Lysxia** [comment]
Right, other than explaining it in terms of sums and products, which is literally what `Generic` does, I don't know.
